lib/generator.js

```diff
@@ -86,14 +87,27 @@ class Generator {
    * @param {Boolean} [options.install=false] Install the template and its dependencies, even when the template has already been installed.
    * @param {Boolean} [options.debug=false] Enable more specific errors in the console. At the moment it only shows specific errors about filters. Keep in mind that as a result errors about template are less descriptive.
    * @param {Object<String, String>} [options.mapBaseUrlToFolder] Optional parameter to map schema references from a base url to a local base folder e.g. url=https://schema.example.com/crm/  folder=./test/docs/ .
+   * @param {Object} [options.registry]           Optional parameter with private registry configuration
+   * @param {String} [options.registry.url]       Parameter to pass npm registry url
+   * @param {String} [options.registry.username]  Optional parameter to pass npm registry username
+   * @param {String} [options.registry.password]  Optional parameter to pass npm registry base64 encoded password
+   * @param {String} [options.registry.token]     Optional parameter to pass npm registry auth token
    */
-  constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false, mapBaseUrlToFolder = {} } = {}) {
-    const invalidOptions = getInvalidOptions(GENERATOR_OPTIONS, arguments[arguments.length - 1] || []);
+  
+constructor(templateName, targetDir, { templateParams = {}, entrypoint, noOverwriteGlobs, disabledHooks, output = 'fs', forceWrite = false, install = false, debug = false, mapBaseUrlToFolder = {}, registry = {}} = {}) {
+    const options = arguments[arguments.length - 1];
+    const invalidOptions = getInvalidOptions(GENERATOR_OPTIONS, options || []);
     if (invalidOptions.length) throw new Error(`These options are not supported by the generator: ${invalidOptions.join(', ')}`);
+    if (options?.registry) {
+       const invalidRegOptions = getInvalidOptions(REGISTRY_OPTIONS, options.registry || []);
+       this.isInvalidRegOptions(invalidRegOptions)
```

Comment:
I like it, but it will not be consistent with validation of options that you have above. Maybe refactoring of above parts in this PR makes sense as well, so we get `this.isInvalidOptions(invalidRegOptions)` ?

Reply:
Resolved @derberg